### PR TITLE
🐛 Add a schema to the gRPC interceptor

### PR DESCRIPTION
**Pull request by fuzzybinary**
### What and why?

Apparently to connect RUM resources properly to distributed tracing, we need to include a URL schema. For now, we're adding http / https to all calls that do not include a schema.

### Review checklist

- [x] This pull request has appropriate unit and / or integration tests 
- [x] This pull request references a Github or JIRA issue

### CI Configuration (optional)
- [ ] Run unit tests
- [x] Run integration tests

### CODE REVIEW

File: packages/datadog_grpc_interceptor/test/datadog_grpc_interceptor_test.dart [hunk 1]
```diff
@@ -39,77 +39,246 @@ void main() {
...
+      expect(call.clientMetadata!['x-datadog-parent-id'], isNotNull);
+      expect(
+          BigInt.tryParse(
+              call.clientMetadata!['x-datadog-parent-id'] as String),
+          isNotNull);
+      expect(call.clientMetadata!['x-datadog-origin'], 'rum');
+      expect(call.clientMetadata!['x-datadog-sampling-priority'], '1');
+    });
+
+    test('Interceptor does not send traces for when shouldSample returns false',
```
Comment (0xnm) [inline packages/datadog_grpc_interceptor/test/datadog_grpc_interceptor_test.dart:159]: seems test name has a mistake?

**Review by 0xnm — APPROVED** [review]

**Review by xgouchet — APPROVED** [review]

File: packages/datadog_grpc_interceptor/test/datadog_grpc_interceptor_test.dart (hunk 1 above)
Comment (fuzzybinary) [reply packages/datadog_grpc_interceptor/test/datadog_grpc_interceptor_test.dart:159]: Is it just that extra "for" or am I missing something else?

Comment (0xnm) [reply packages/datadog_grpc_interceptor/test/datadog_grpc_interceptor_test.dart:159]: yes, also maybe it should say `traces metadata` by looking on the title of the previous test.

**Comment by acoutts** [comment]
is this one released on pub.dev yet?

**Comment by fuzzybinary** [comment]
@acoutts Yes, this should have been in the 1.0.0-beta.1 release for [this package](https://pub.dev/packages/datadog_grpc_interceptor)

**Comment by acoutts** [comment]
Hm i seem to get this error when I try to run using the latest published version of the package:
```
    1 warning generated.
: Error: Too few positional arguments: 3 required, 1 given.
../…/src/tracking_http_client.dart:114
          datadogSdk.internalLogger.sendToDatadog(
                                                 ^
: Error: Too few positional arguments: 3 required, 1 given.
../…/src/tracking_http_client.dart:134
              datadogSdk.internalLogger.sendToDatadog(
                                                     ^
: Error: Too few positional arguments: 3 required, 1 given.
../…/src/tracking_http_client.dart:316
          datadogSdk.internalLogger.sendToDatadog(
                                                 ^
: Error: Too few positional arguments: 3 required, 1 given.
../…/src/tracking_http_client.dart:462
          datadogSdk.internalLogger.sendToDatadog(
                                                 ^
    Failed to package /Users/andrew.coutts/Projects/app/packages/app.
    Command PhaseScriptExecution failed with a nonzero exit code
    note: Building targets in dependency order
```

**Comment by acoutts** [comment]
I think something changed in flutter 3.3.0 for HttpClient

**Comment by fuzzybinary** [comment]
That's strainge... can you post what version solving gave you for all of the Datadog libraries?

All up to date published packages of datadog_tracking_http_client (1.0.1 for pre-Flutter 3.0 and 1.1.0 for post-Flutter 3.0) should be using datadog_flutter_plugin ^1.0.0, which should require 3 arguments to that method.

There was a change in the dart:io `HttpClient` in Flutter 3 / Dart 2.18, but I don't know of any changes since then.

**Pull request merged**